engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2178,6 +2178,19 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
 
   }
 
+  private ApplicationReport findApplicationReportFromAppNameOrId(String appNameOrId)
+  {
+    ApplicationReport app = null;
+
+    if (getApplication(appNameOrId) != null) {
```

Comment:
app = getApplication(appNameOrId)
if (app == null) {
     app = getApplicationByName(appNameOrId)
}

To avoid calling getApplication and getApplicationByName twice.